lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -286,17 +274,17 @@ def test_has_access_with_preview_mode(self):
         self.assertFalse(bool(access.has_access(self.student, 'staff', self.course, course_key=self.course.id)))
         self.assertFalse(bool(access.has_access(self.student, 'load', self.course, course_key=self.course.id)))
 
-        # User should be able to preview when masquerade.
+        # When masquerading is true, user should not be able to access staff content
         with patch('courseware.access.is_masquerading_as_student') as mock_masquerade:
             mock_masquerade.return_value = True
-            self.assertTrue(
+            self.assertFalse(
```

Comment:
This check i mean `has_admin_access_to_course or is_masquerading_as_student(user, course_key)`